File: Civitas-Java/Civitas/src/main/java/civitas/SorpresaESPECULADOR.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 * @author elena
 * @date 13/12/2019
 */
public class SorpresaESPECULADOR extends Sorpresa{
    
    SorpresaESPECULADOR(int valor, String texto){   
        super(valor, texto);
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            
            JugadorEspeculador nuevo_jugador = new JugadorEspeculador(todos.get(actual));
            todos.remove(todos.get(actual));
            todos.add(actual, nuevo_jugador);
        }
    }
    
    @Override
    public String toString(){
        return super.toString() + 
               "\nTipo Sorpresa: Jugador Especulador";
    }
}

File: Civitas-Ruby/lib/sorpresa_especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "tipo_sorpresa"
require_relative "mazo_sorpresas"
require_relative "tablero"
require_relative "jugador"
require_relative "jugador_especulador"

module Civitas
  class SorpresaEspeculador < Sorpresa
    def initialize(valor, texto)
      super(valor, texto)
    end
    
    @override
    def aplicarajugador(actual, todos)
      if jugadorcorrecto(actual, todos)
        informe(actual, todos)
        
        jugador = JugadorEspeculador.new(todos[actual])
        todos.delete(todos[actual])
        todos[actual] << jugador
      end
    end
    
    @override
    def to_s
      super.to_s + 
      '\nTipo Sorpresa: Jugador Especulador'
    end
  end
end
